### PR TITLE
Test suite refactor

### PR DESCRIPTION
**Pull request by MatteoNardi**
This PR is a tentative refactor of the test suite.
Tests are split in a different binary, which can be invoked with sudo by using a xtask.

Fixes https://github.com/Exein-io/pulsar/issues/3

### CODE REVIEW

**Comment by MatteoNardi** [comment]
All the test were ported to the new system. I'll work on documentation and general cleanup, than it will be ready for review.

**Comment by MatteoNardi** [comment]
This PR is now ready for review.

- I've split the test suite in a different binary. See [README](https://github.com/Exein-io/pulsar/tree/test-suite/test-suite).
- I've added xtasks for every pulsar binary which need to be run as sudo. [See README](https://github.com/Exein-io/pulsar/tree/test-suite/xtask)
  - I've made `probe` a binary to make it more coherent with the others.
- `cargo test` doesn't require sudo anymore. I've re-enabled `validatron` tests.
- All documentation references in the repository should have been updated.

Some notes:

- I've decided to get rid of [inventory](https://github.com/dtolnay/inventory) for tests collection as I don't think the added value justified the added complexity. Every module has a test entrypoint which must be listed in the `test-suite` binary.

**Review by banditopazzo — CHANGES_REQUESTED** [review]

File: bpf-common/src/test_runner.rs [hunk 1]
```diff
@@ -75,128 +121,212 @@ impl<T: Display> TestRunner<T> {
...
-        let found = self.events.iter().map(predicate).any(|x| x);
-        if !found {
-            panic!("event not found among {} analyzed", self.events.len());
+impl<T> Clone for TestSender<T> {
+    fn clone(&self) -> Self {
+        Self {
+            tx: self.tx.clone(),
         }
     }
+}
```
[the remark this answers is not in the document]
Comment (MatteoNardi) [reply bpf-common/src/test_runner.rs:144]: The derive implementation doesn't support generics

**Comment by MatteoNardi** [comment]
I've rebased to main and added https://github.com/Exein-io/pulsar/pull/25/commits/2b3e58454b91f689329490850a2b83112f5c13e3

**Review by banditopazzo — APPROVED** [review]

**Pull request merged**